src/fsi/console.fs

```diff
@@ -50,60 +65,88 @@ type internal Options() =
 
     let mutable root = ""
 
-    member _.Root with get() = root and set(v) = (root <- v)
+    member _.Root
+        with get () = root
+        and set (v) = (root <- v)
 
 /// Cursor position management
 
 module internal Utils =
 
-    let guard(f) =
-        try f()
-        with e ->
-             FSharp.Compiler.DiagnosticsLogger.warning(Failure(sprintf "Note: an unexpected exception in fsi.exe readline console support. Consider starting fsi.exe with the --no-readline option and report the stack trace below to the .NET or Mono implementors\n%s\n%s\n" e.Message e.StackTrace))
+    let guard (f) =
+        try
+            f ()
+        with
+        | e ->
+            warning (
+                Failure(
+                    sprintf
+                        "Note: an unexpected exception in fsi.exe readline console support. Consider starting fsi.exe with the --no-readline option and report the stack trace below to the .NET or Mono implementors\n%s\n%s\n"
+                        e.Message
+                        e.StackTrace
+                )
+            )
```

Comment:
These expansions for diagnostics and `sprintf` calls look like they're going to be painful.

Most `sprintf` will be better as interpolated strings, but still, thats 1 line --> 8 lines here